### ISSUE TITLE
Condition !Equals function does not support !Select or !Join as arguments

### ISSUE THREAD

**Issue by phene** [aws-cloudformation/cfn-python-lint#881]
*cfn-lint version: 0.20.0*

Example valid CF template chunk:

```yaml
Parameters:
  SomeList:
    Type: CommaDelimitedList

Conditions:
  ListStartsWithFoo: !Equals [ !Select [ 0, !Ref SomeList ], Foo ]  
```

Fails with `E8003 Fn::Equals element must be a string, Ref, or a Fn::FindInMap`. It should accept any intrinsic function that results in a string, including `!Select` and`!Join`.

**Comment by kddejong** [comment]
Can you try version 0.20.1?  This should be fixed in that release.

**Comment by phene** [comment]
Oh, looks like you did fix it. Looking at the [fix](https://github.com/aws-cloudformation/cfn-python-lint/commit/868b2262349c41d76a38dd127faa2860e2d66ec1), though, it doesn't seem like `Fn::Split` should be included since that will always return a list.

**Comment by kddejong** [comment]
Very fair point :(  I'll get that removed.  Thanks for noticing that.
